semgrep/tests/e2e/test_ignores.py

```diff
@@ -1,40 +1,14 @@
-from subprocess import CalledProcessError
+from pathlib import Path
 
-import pytest
+from ..conftest import TESTS_PATH
 
 
-def test_regex_rule__nosemgrep(run_semgrep_in_tmp, snapshot):
-    snapshot.assert_match(
-        run_semgrep_in_tmp(
-            "rules/regex-nosemgrep.yaml", target_name="basic/regex-nosemgrep.txt"
-        )[0],
-        "results.json",
-    )
-
-
-def test_nosem_rule(run_semgrep_in_tmp, snapshot):
-    snapshot.assert_match(run_semgrep_in_tmp("rules/nosem.yaml")[0], "results.json")
-
-
-def test_nosem_rule_unicode(run_semgrep_in_tmp, snapshot):
-    snapshot.assert_match(
-        run_semgrep_in_tmp(
-            "rules/nosem-unicode.yaml", target_name="advanced_nosem/nosem-unicode.py"
-        )[0],
-        "results.json",
+def test_semgrepignore(run_semgrep_in_tmp, tmp_path, snapshot):
+    (tmp_path / ".semgrepignore").symlink_to(
+        Path(TESTS_PATH / "e2e" / "targets" / "ignores" / ".semgrepignore").resolve()
     )
 
-
-def test_nosem_rule__invalid_id(run_semgrep_in_tmp, snapshot):
-    with pytest.raises(CalledProcessError) as excinfo:
-        run_semgrep_in_tmp("rules/nosem.yaml", target_name="nosem_invalid_id")
-    assert excinfo.value.returncode == 2
-    snapshot.assert_match(excinfo.value.stderr, "error.txt")
-    snapshot.assert_match(excinfo.value.stdout, "error.json")
-
-
-def test_nosem_rule__with_disable_nosem(run_semgrep_in_tmp, snapshot):
     snapshot.assert_match(
-        run_semgrep_in_tmp("rules/nosem.yaml", options=["--disable-nosem"])[0],
+        run_semgrep_in_tmp("rules/eqeq-js.yaml", target_name="ignores")[0],
```

Comment:
We should probably test each of the parser cases (file exclusion, directory exclusion, directory exclusion with a `/` in front, include directive resolution, etc.) but I won't block approval on that

Reply:
You can probably port https://github.com/returntocorp/bento/blob/master/tests/test_fignore.py

Reply:
I'll just make a separate PR with more extensive tests. I think what we have is ok for now, considering this is mostly copied directly from semgrep-action, which doesn't seem to have major issues with .semgrepignore